zhaquirks/tuya/__init__.py

```diff
@@ -54,37 +72,40 @@ class TuyaManufClusterAttributes(TuyaManufCluster):
 
     def handle_cluster_request(self, tsn: int, command_id: int, args: Tuple) -> None:
         """Handle cluster request."""
-        if command_id not in (0x0001, 0x0002):
+        if not (
+            command_id in (0x0001, 0x0002)
+            and len(args) == 1
+            and isinstance(args[0], TuyaManufCluster.Command)
```

Comment:
aren't these conditions a little redundant?

if `command_id in (0x0001, 0x0002)` is True, then `isinstance(args[0], TuyaManufCluster.Command)` is True automatically, so I think the last condition could be omitted. and by definition if it is a 0x0001 or 0x0002 command, then `len(args) == 1`

Reply:
Are you sure about that ? I was afraid that no real typing is done at run time. I seem to remember that I had the `command_id` test at one point yet zoic was still getting the AttributeError thing. And I have been trained into never accessing a list element if the list is not long enough.

I can change it back of course.

Reply:
Yeah, but for this cluster on line 64--67 
```
    manufacturer_client_commands = {
        0x0001: ("get_data", (Command,), True),
        0x0002: ("set_data_response", (Command,), True),
    }
```

You define only two commands: 0x0001 & 0x0002 and each command has a schema of `(Command,)` -- so if you the execution got to `handle_cluster_request()` that means that the deserialization of the ZCL frame was successful. 
Since the deserialization was successful **and** `command_id` is either 0x0001 or 0x0002, then we can assume there's only one argument, because the schema for those commands has only one 1 argument and that is `Command` class type. 
In other words for that specific cluster, as long as `command_id in (0x0001, 0x0002)` then the `*args` contains only one element and it is an instance of `Command` class

Reply:
So I suppose that the report from the user was from a version where I still didn't test for `command_id` then. I was under the impression that he still reported the problem *after* I added the check, but since you confirm this is not possible, I will revert back to just test the `command_id`

Reply:
If it was my i have not updating the last version with the new tests applied.
If you need i installing the latest commit and redoing the logs.